smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/lang/DocumentationPreprocessor.kt

```diff
@@ -21,17 +26,226 @@ class DocumentationPreprocessor : KotlinIntegration {
         return transformer.mapTraits(model) { _, trait ->
             when (trait) {
                 is DocumentationTrait -> {
-                    val docs = sanitize(trait.value)
+                    // There's definitely some improperly escaped HTML characters within preformat blocks in existing
+                    // models. Ensure we strip those now, the parser is VERY forgiving and will mistreat any sequences
+                    // of characters that happen to form tags as such.
+                    val sanitizedDoc = trait.value
+                        .applyWithin("<code>", "</code>", String::escapeHtml)
```

Comment:
what if there are multiple `<code>` blocks in a single trait? looks like the `applyWithin` will find first occurrence.

Reply:
Oh nevermind I missed that it was recursive